src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -829,36 +816,38 @@ bool FancyZones::OnSnapHotkey(DWORD vkCode) noexcept
 
 void FancyZones::HandleVirtualDesktopUpdates(HANDLE fancyZonesDestroyedEvent) noexcept
 {
+    HKEY virtualDesktopsRegKey = VirtualDesktopUtils::GetVirtualDesktopsRegKey();
+    if (!virtualDesktopsRegKey)
+    {
+        return;
+    }
     HANDLE regKeyEvent = CreateEvent(nullptr, FALSE, FALSE, nullptr);
     HANDLE events[2] = { regKeyEvent, fancyZonesDestroyedEvent };
     while (1)
     {
-        if (RegNotifyChangeKeyValue(m_virtualDesktopsRegKey, TRUE, REG_NOTIFY_CHANGE_LAST_SET, regKeyEvent, TRUE) != ERROR_SUCCESS)
+        if (RegNotifyChangeKeyValue(virtualDesktopsRegKey, TRUE, REG_NOTIFY_CHANGE_LAST_SET, regKeyEvent, TRUE) != ERROR_SUCCESS)
         {
-            return;
+            break;
         }
         if (WaitForMultipleObjects(2, events, FALSE, INFINITE) != (WAIT_OBJECT_0 + 0))
         {
             // if fancyZonesDestroyedEvent is signalized or WaitForMultipleObjects failed, terminate thread execution
-            return;
-        }
-        std::vector<GUID> ids{};
-        if (VirtualDesktopUtils::GetVirtualDekstopIds(m_virtualDesktopsRegKey, ids))
-        {
-            std::unordered_set<GUID> idSet(std::begin(ids), std::end(ids));
-            RegisterVirtualDesktopUpdates(idSet);
+            break;
         }
+        PostMessage(m_window, WM_PRIV_VDUPDATE, 0, 0);
     }
+    VirtualDesktopUtils::CloseVirtualDesktopsRegKey();

```

Comment:
let's use `wil::unique_hkey` for `virtualDesktopsRegKey` so we don't need this manually-invoked clean up

Reply:
Done.